### ISSUE TITLE
Multi-line prompt text offset issue

### ISSUE THREAD

**Issue by jibbers42** [sbt/sbt#625]
When using a multi-line prompt, such as below, text will appear offset to the right and duplicated when up arrowing through history or doing tab completion. Also, at this point the text cannot be deleted.

I'm a huge fan of the multi-line prompt as it give a consistent text entry location and also clearly delineates the command history.

I'm using sbt 0.12.1 on Ubuntu 12.04

``` scala
object ShellPrompt {
  object devnull extends ProcessLogger {
    def info (s: => String) {}
    def error (s: => String) { }
    def buffer[T] (f: => T): T = f
  }
  def currBranch = (
    ("git status -sb" lines_! devnull headOption)
      getOrElse "-" stripPrefix "## "
  )

  val buildShellPrompt = {
    (state: State) => {
      val currProject = Project.extract (state).currentProject.id
      """
%s:%s:%s
sbt> """.format (
        currProject, currBranch, VersionSettings.buildVersion
      )
    }
  }
}
```

**Comment by harrah** [comment]
I would guess that JLine gets confused.  A possible workaround is to have sbt split the string into lines and print out all but the last one itself.  The last one would be given to JLine as usual.  A pull request is welcome.  The place to modify it is here:

https://github.com/harrah/xsbt/blob/0.13/util/complete/src/main/scala/sbt/LineReader.scala#L44

**Comment by harrah** [comment]
Fixed in f7031502e4bb31d55d514e73613f6f4761232808.
